packages/dashboard-core-plugins/src/panels/Panel.tsx

```diff
@@ -151,6 +155,11 @@ class Panel extends PureComponent<PanelProps, PanelState> {
     if (prevProps.componentPanel == null && componentPanel != null) {
       glEventHub.emit(PanelEvent.MOUNT, componentPanel);
     }
+
+    this.setState({
```

Comment:
Shouldn't we check this in `componentDidMount` instead?

Reply:
Maybe. I wasn't sure if there would end up being a case where the ref or parent container could change, so I put it in update for max safety. Should be fine in `componentDidMount` though and it's not like it's an app breaking bug if we later find out it does need to be in `componentDidUpdate`